src/distance_finder.cpp

```diff
@@ -133,6 +141,34 @@ class DistanceFinder {
         return false;
     }
 
+    bool validateRadius(std::string &prop_label, double radius) {
+        if ((prop_label == "red_marker" || prop_label == "green_marker") && (radius>marker_radius_p-marker_radius_range_p && radius<marker_radius_p+marker_radius_range_p)) {
+            return true;
+        }
+        if ((prop_label == "blue_buoy") && (radius>buoy_radius_lg_p-buoy_lg_radius_range_p && radius<buoy_radius_lg_p+buoy_lg_radius_range_p)) {
+            return true;
+        }
+        if ((prop_label == "black_buoy") && (radius>buoy_radius_sm_p-buoy_sm_radius_range_p && radius<buoy_radius_sm_p+buoy_sm_radius_range_p)) {
+            return true;
+        }
+        if ((prop_label == "red_buoy" || prop_label == "green_buoy" || prop_label == "yellow_buoy")) {
+            if(!check_small_buoys_p && ((radius>(((buoy_radius_sm_p+buoy_radius_lg_p)/2)-buoy_lg_radius_range_p)) && (radius<(((buoy_radius_sm_p+buoy_radius_lg_p)/2)+buoy_lg_radius_range_p)))) {
+                return true;
+            }
+            if (radius>buoy_radius_lg_p-buoy_lg_radius_range_p && radius<buoy_radius_lg_p+buoy_lg_radius_range_p) {
```

Comment:
I think you need to add `check_small_buoys` as a condition for the assigning small or large states if statements.